scipy/stats/_continued_fraction.py

```diff
@@ -0,0 +1,399 @@
+        nfev : int array
+            The number of terms that were evaluated.
+
+    Notes
+    -----
+    A generalized continued fraction is an expression of the form
+
+    .. math::
+
+        b_0 + \frac{a_1}{b_1 + \frac{a_2}{b_2 + \frac{a_3}{b_3 + \cdots}}}
+
+    Successive "convergents" approximate the infinitely recursive continued fraction
+    with a finite number of terms :math:`a_n` and :math:`b_n`, which are provided
+    by callables `a` and `b`, respectively. This implementation follows the modified
+    Lentz algorithm ([1]_, [2]_) to evaluate successive convergents until a
+    termination condition is satisfied.
+
+    References
+    ----------
+    .. [1] Press, William H., and Saul A. Teukolsky. "Evaluating continued fractions
+           and computing exponential integrals." Computers in Physics 2.5 (1988): 88-89.
+    .. [2] Lentz's algorithm. Wikipedia.
+           https://en.wikipedia.org/wiki/Lentz%27s_algorithm
+    .. [3] Continued fraction. Wikipedia.
+           https://en.wikipedia.org/wiki/Continued_fraction
+    .. [4] Generalized continued fraction. Wikipedia.
+           https://en.wikipedia.org/wiki/Generalized_continued_fraction
+
+    Examples
+    --------
+    The "simple continued fraction" of :math:`\pi` is given in [3]_ as
+
+    .. math::
+
+        3 + \frac{1}{7 + \frac{1}{15 + \frac{1}{1 + \cdots}}}
+
+    where the :math:`b_n` terms follow no obvious pattern:
+
+    >>> b = [3, 7, 15, 1, 292, 1, 1, 1, 2, 1, 3, 1]
+
+    and the :math:`a_n` terms are all :math:`1`.
+    In this case, all the terms have been precomputed, so we call `_continued_fraction`
+    with simple callables which simply return the precomputed coefficients:
+
+    >>> res = _continued_fraction(a=lambda n: 1, b=lambda n: b[n], maxiter=len(b) - 1)
+    >>> (res.f - np.pi) / np.pi
+    np.float64(7.067899292141148e-15)
+
+    A generalized continued fraction for :math:`\pi` is given by:
+
+    .. math::
+
+        3 + \frac{1^2}{6 + \frac{3^2}{6 + \frac{5^2}{6 + \cdots}}}
+
+    We define the coefficient callables as:
+
+    >>> def a(n):
+    ...     return (2*n - 1)**2
+    >>>
+    >>> def b(n):
+    ...     if n == 0:
+    ...         return 3
+    ...     else:
+    ...         return 6
+
+    Then the continued fraction can be evaluated as:
+
+    >>> from scipy.special._continued_fraction import _continued_fraction
+    >>> res = _continued_fraction(a, b)
+    >>> res
+         success: False
+          status: -2
+               f: 3.1415924109719846
+             nit: 100
+            nfev: 101
+
+    Note that the requested tolerance was not reached within the (default)
+    maximum number of iterations because it converges very slowly.
+    An expression that converges more rapidly is expressed as the difference
+    between two continued fractions. We will compute both of them in one
+    vectorized call to `_continued_fraction`.
+
+    >>> import numpy as np
+    >>> u, v = 5, 239
+    >>>
+    >>> def a(n, a1, _):
```

Comment:
Sorry, what specifically?

Reply:
Handling the situation where `a` and `b` need different arguments as there aren't many functions in SciPy that take two callables

Reply:
Hmm I still don't really see anything to point out. It's documented that all `args` are passed to both functions. Maybe you noticed that I named the the variable `_` because it isn't used? It's pretty common to name an output `_` when unpacking an iterable and the value won't be used. I could rename this to `uv` and maybe then it wouldn't look so notable? It's not a convention that anyone has to follow, though.

Reply:
Happy just to leave as after reading again